src/Product/Importer.php

```diff
@@ -354,4 +357,27 @@ public function reconcile(array $identifiersToReconcileWith): void
             $this->dispatchPostEvent($product, 'update');
         }
     }
+
+    private function disableOldParentProductIfItHasNotAnyVariants(string $identifier, ProductInterface $product): void
+    {
+        $oldParentProduct = $this->productRepository->findOneByCode($identifier);
+        if ($oldParentProduct === null) {
+            return;
+        }
+        if ($oldParentProduct === $product) {
+            return;
+        }
+        if ($oldParentProduct->getVariants()->count() !== 1) {
+            return;
+        }
+        $productVariant = $oldParentProduct->getVariants()->first();
+        if (!$productVariant instanceof ProductVariantInterface) {
+            return;
+        }
+        if ($productVariant->getCode() !== $identifier) {
+            return;
+        }
```

Comment:
Il metodo si chiama `disableOldParentProductIfItHasNotAnyVariants` ma poi però lo si disabilita solo se ha 1 variante con quell'identificatore passato.

Come mai non avete preferito questa strada andando poi a spostare la chiamata del metodo dopo che la variante è stata associata al nuovo parent?

```suggestion
        if (!$oldParentProduct->getVariants()->isEmpty()) {
            return;
        }
```

Reply:
Perché è vero che la variante viene associata al nuovo Product, ma nel vecchio Product la collezioni di varianti rimane invariata. Il metodo setProduct sulla ProductVariant dovrebbe essere così:
```php
  public function setProduct(?ProductInterface $product): void
  {
+    if ($this->product !== $product) {
+        $this->product->removeVariant($this);
+    }
      $this->product = $product;
  }
```

Reply:
Ok